### ISSUE TITLE
How does Keras-YOLOv7 deal with conv2d layer padding bug?

### ISSUE THREAD

**Issue by abhaskumarsinha** [keras-team/keras-cv#2379]
### Current Behavior:

Mismatch of the elements in the edges and corners in Keras Conv2D Layer when comparing the conv2d of YOLOv9 that was implemented on PyTorch.

### Expected Behavior:

No mismatch.

### Anything else:

I believe the code for YOLOv9 was extended from YOLOv7 that has already been implemented in KerasCV. How did they not encounter this issue? Is there any problem on my side?

Linked: https://github.com/keras-team/keras/issues/19291#issue-2180964867

**Comment by innat** [comment]
YOLO-V7 https://github.com/keras-team/keras-cv/issues/1179

**Comment by abhaskumarsinha** [comment]
@innat I see, the padding of the element in the edges are manually implemented here.

**Comment by abhaskumarsinha** [comment]
@innat Ah, the problem with that implementation is that I won't get to use to switch to backends after using that...
If I switch from the TensorFlow backend to the PyTorch backend using that code, the code would break. There's no point in implementing that if that doesn't get fixed! If that doesn't remain backend invariant, there's no point of using Keras at all then!

Let me post the code to the main issue so that gets attention.
